app/src/test/java/com/mho/training/TestDI.kt

```diff
@@ -0,0 +1,153 @@
+package com.mho.training
+
+import com.example.android.data.repositories.*
+import com.example.android.data.sources.*
+import com.example.android.domain.*
+import com.example.android.domain.result.DataResult
+import com.example.android.testshared.*
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.flowOf
+import org.koin.core.context.startKoin
+import org.koin.core.module.Module
+import org.koin.dsl.module
+
+fun initMockedDi(vararg modules: Module) {
+    startKoin {
+        modules(listOf(mockedAppModule, mockedDataModule) + modules)
+    }
+}
+
+private val mockedAppModule = module {
+    single<LocalMovieDataSource> { FakeLocalMovieDataSource() }
+    single<RemoteMovieDataSource> { FakeRemoteMovieDataSource() }
+    single<RemoteCreditDataSource> { FakeRemoteCreditDataSource() }
+    single<RemoteKeywordDataSource> { FakeRemoteKeywordDataSource() }
+    single<RemotePersonDataSource> { FakeRemotePersonDataSource() }
+    single<RemoteReviewDataSource> { FakeRemoteReviewDataSource() }
+    single<RemoteTrailerDataSource> { FakeRemoteTrailerDataSource() }
+    single<LocationDataSource> { FakeLocationDataSource() }
+    single<PermissionChecker> { FakePermissionChecker() }
+    single { Dispatchers.Unconfined }
+}
+
+private val mockedDataModule = module {
+    factory { CreditRepository(get()) }
+    factory { KeywordRepository(get()) }
+    factory { MovieRepository(get(), get(), get()) }
+    factory { PersonRepository(get()) }
+    factory { RegionRepository(get(), get()) }
+    factory { ReviewRepository(get()) }
+    factory { TrailerRepository(get()) }
+}
+
+val defaultFakeCredits = listOf(
```

Comment:
Have you considered to put those fake objects into a sharetest module?

Reply:
It's a great idea! I'll check it.